### PR TITLE
8268522: InstanceKlass::can_be_verified_at_dumptime() returns opposite value

### PR DESCRIPTION
**Pull request by calvinccheung**
Please review this small patch:

- correct the return value of `InstanceKlass::can_be_verified_at_dumptime()` to match the function's name;
- modify the call sites accordingly.

- [x] mach5 tiers 1,2,3

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8268522](https://bugs.openjdk.java.net/browse/JDK-8268522): InstanceKlass::can_be_verified_at_dumptime() returns opposite value

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Yumin Qi](https://openjdk.java.net/census#minqi) (@yminqi - **Reviewer**) ⚠️ Review applies to c43bcb07f9e25d2ca3c2ada26089131338aae54e
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17 pull/31/head:pull/31` \
`$ git checkout pull/31`

Update a local copy of the PR: \
`$ git checkout pull/31` \
`$ git pull https://git.openjdk.java.net/jdk17 pull/31/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 31`

View PR using the GUI difftool: \
`$ git pr show -t 31`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17/pull/31.diff">https://git.openjdk.java.net/jdk17/pull/31.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/oops/klassVtable.cpp [hunk 1]
```diff
@@ -1094,8 +1094,8 @@ void itableMethodEntry::initialize(InstanceKlass* klass, Method* m) {
 #ifdef ASSERT
   if (MetaspaceShared::is_in_shared_metaspace((void*)&_method) &&
      !MetaspaceShared::remapped_readwrite() &&
-     !m->method_holder()->can_be_verified_at_dumptime() &&
-     !klass->can_be_verified_at_dumptime()) {
+     m->method_holder()->verified_at_dump_time() &&
+     klass->verified_at_dump_time()) {
```
Comment (dholmes-ora) [inline src/hotspot/share/oops/klassVtable.cpp:1098]: This change seems to go beyond the simple logic negation. Why the change here?

**Review by dholmes-ora — APPROVED** [review]
Hi Calvin,

The logic negation seems fine.

The change to shared_old_class also seems fine in general.

But there is one change that I'm querying as it seems a functional change.

Thanks,
David

File: src/hotspot/share/oops/klassVtable.cpp (hunk 1 above)
Comment (calvinccheung) [reply src/hotspot/share/oops/klassVtable.cpp:1098]: Hi David,

Thanks for your review!

The block of code you mentioned is being called during runtime. So it is more consistent to call `verified_at_dump_time()`.

Thanks,
Calvin

**Review by yminqi — APPROVED** [review]
LGTM.

File: src/hotspot/share/oops/instanceKlass.cpp [hunk 2]
```diff
@@ -2402,9 +2402,9 @@ void InstanceKlass::metaspace_pointers_do(MetaspaceClosure* it) {
 
 void InstanceKlass::remove_unshareable_info() {
 
-  if (can_be_verified_at_dumptime()) {
-    // Set the old class bit.
-    set_is_shared_old_klass();
+  if (!can_be_verified_at_dumptime()) {
+    // Clear the _verified_at_dump_time bit.
+    clear_verified_at_dump_time();
```
Comment (iklam) [inline src/hotspot/share/oops/instanceKlass.cpp:2407]: I think it's better to change this to 

```
if (can_be_verified_at_dumptime()) {
  // Remember this so we can avoid walking the hierarchy at runtime.
  set_verified_at_dump_time();
```

and do not set this bit inside `Klass::Klass()`. This way, all the code that changes this bit are kept in a single place.

**Review by iklam — CHANGES_REQUESTED** [review]
Just one request for change. The rest looks good to me.

**Comment by calvinccheung** [comment]
> LGTM.

Thanks Yumin.

File: src/hotspot/share/oops/instanceKlass.cpp (hunk 2 above)
Comment (calvinccheung) [reply src/hotspot/share/oops/instanceKlass.cpp]: Thanks Ioi.

I've pushed an update with your suggestion.

**Review by iklam — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]
Update looks good.

Thanks,
David
